core/common/src/main/java/alluxio/util/executor/DeltaQueue.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util.executor;
+
+import javax.annotation.Nullable;
+
+/**
+ * A delay queue with extra supports for passing the head element delay time.
+ *
+ * @param <T> the type of the queue objects
+ */
+public class DeltaQueue<T> {
```

Comment:
could this class be replaced with a priority queue?

Reply:
I used a combination with `priorityQueue` and `mPastTime` in the new `ControllableQueue`.
`mPastTime` record the time that has been past (by jumping to a future time).